editor/script/cursor.gd

```diff
@@ -2,11 +2,54 @@ extends Node3D
 
 class_name BuildableCursor
 
+@export var overlap_label: Label
+
+var overlap_area: Area3D = null :
+	get:
+		return overlap_area
+	set(new_area):
+		# Remove the previous overlap_area signal connections
+		if overlap_area != null:
+			overlap_area.disconnect("area_entered", _on_overlap_begin)
+			overlap_area.disconnect("area_exited", _on_overlap_stop)
+			
+		overlap_area = new_area
+		
+		# The curosr does not need to be detectable by other areas
+		overlap_area.monitorable = false
+		# The cursor must be able to detect other areas
+		overlap_area.monitoring = true
+		
+		# Reset the collision mask and layer
+		overlap_area.collision_layer = 0
+		overlap_area.collision_mask = 0
+		
+		# The cursor must look for areas on layer 2 
+		overlap_area.set_collision_mask_value(2, true)
+		
+		# Connect colision signals to the appropriate functions
+		overlap_area.connect("area_entered", _on_overlap_begin)
+		overlap_area.connect("area_exited", _on_overlap_stop)
+
 @export var cursor_mesh: Buildable
 @export var old_mesh: Buildable
 
+var unplacable: bool = false :
```

Comment:
Personally I think its better to have positive variable names over negative. `is_placeable `over a `is_not_placeable`. Helps with readability when you have to invert `!unplaceable` in your head.

Reply:
I agree, updated👍